client/src/main/java/no/unit/nva/model/client/ResourceDtoBuilder.java

```diff
@@ -0,0 +1,25 @@
+package no.unit.nva.model.client;
+
+import no.unit.nva.model.external.ResourceDto;
+
+import java.net.URI;
+
+public final class ResourceDtoBuilder {
```

Comment:
I would suggest a builder that re-uses the setters of the ResourceDto. The setters can be non-public. The the logic validating the input will be in one place.

```
public ResourceDtoBuilder {

private ResourceDto resourceDto;

public ResourceBuilder withField(Field field){
 resourceDto.setField(field);
return this;
}

public ResrouceDto build(){
return resouceDto;
}
```

Reply:
It is also more economical regarding number of objects being built